client/state/sites/hooks/use-site-global-styles-status.ts

```diff
@@ -1,29 +1,90 @@
 import { useSelector } from 'calypso/state';
+import { isUserLoggedIn } from 'calypso/state/current-user/selectors';
 import { getSite } from 'calypso/state/sites/selectors';
 import { getSelectedSiteId } from 'calypso/state/ui/selectors';
 
 export type GlobalStylesStatus = {
 	shouldLimitGlobalStyles: boolean;
 	globalStylesInUse: boolean;
+	globalStylesInPersonalPlan?: boolean;
 };
 
 // While we are loading the Global Styles Info we can't assume that we should limit global styles, or we would be
 // showing notices for paid sites until we fetch the data from the server.
 const DEFAULT_GLOBAL_STYLES_INFO: GlobalStylesStatus = {
 	shouldLimitGlobalStyles: false,
 	globalStylesInUse: false,
+	globalStylesInPersonalPlan: false,
 };
 
-const getGlobalStylesInfoForSite = ( siteId: number | null ): Promise< GlobalStylesStatus > => {
-	if ( siteId === null ) {
+/*
+ * We cannot import `loadExperimentAssignment` directly from 'calypso/lib/explat'
+ * because it runs a side effect that produces an error on SSR contexts.
+ */
+let loadExperimentAssignment = ( experimentName: string ): Promise< ExperimentAssignment > =>
+	Promise.resolve( { experimentName, variationName: null, retrievedTimestamp: 0, ttl: 0 } );
+
+if ( typeof window !== 'undefined' ) {
+	import( 'calypso/lib/explat' )
+		.then( ( module ) => {
+			loadExperimentAssignment = module.loadExperimentAssignment;
+		} )
+		// eslint-disable-next-line @typescript-eslint/no-empty-function
+		.catch( () => {} );
+}
+
+function shouldRunGlobalStylesOnPersonalExperiment(
+	siteId: number | null,
+	userLoggedIn: boolean
+): boolean {
+	// Do not run it on SSR contexts.
+	if ( typeof window === 'undefined' ) {
+		return false;
+	}
+
+	// Always run it if a site has been selected.
+	if ( siteId !== null ) {
+		return true;
+	}
+
+	// Do not run it on the logged-out theme showcase.
+	if ( ! userLoggedIn && window.location.pathname.startsWith( '/theme' ) ) {
+		return false;
+	}
+
+	// Run it by default. Ideally, we should not run it if the user is logged out, but
+	// we cannot rely on the `isUserLoggedIn` selector for users who just signed up
+	// (see pbxNRc-2HR-p2#comment-4607). So, we assume that this hook is not used in
+	// any logged-out context apart from the theme showcase.
+	return true;
+}
+
+const getGlobalStylesInfoForSite = (
+	siteId: number | null,
+	userLoggedIn: boolean = false
+): Promise< GlobalStylesStatus > => {
+	if ( ! shouldRunGlobalStylesOnPersonalExperiment( siteId, userLoggedIn ) ) {
 		return Promise.resolve( {
 			shouldLimitGlobalStyles: true,
 			globalStylesInUse: false,
+			globalStylesInPersonalPlan: false,
 		} );
 	}
 
+	if ( siteId === null ) {
+		return loadExperimentAssignment( 'calypso_plans_global_styles_personal_20240127' ).then(
+			( experimentAssignment ) =>
+				Promise.resolve( {
+					shouldLimitGlobalStyles: true,
+					globalStylesInUse: false,
+					globalStylesInPersonalPlan: experimentAssignment.variationName === 'treatment',
```

Comment:
We have 2 different treatments for this experiment which are: `gs_on_personal` and `gs_on_personal_extra_features`

Reply:
Fixed in https://github.com/Automattic/wp-calypso/pull/98974